addons/survey/models/survey.py

```diff
@@ -0,0 +1,1096 @@
+        result = dict((data['survey_id'][0], data['survey_id_count']) for data in survey_data)
+        for survey in self:
+            survey.tot_start_survey = result.get(survey.id, 0)
+
+    @api.multi
+    def _compute_tot_comp_survey(self):
+        """ Returns the number of completed instances of this survey """
+        survey_data = self.env['survey.user_input'].read_group([('survey_id', 'in', self.ids), ('state', '=', 'done')], ['survey_id'], ['survey_id'])
+        result = dict((data['survey_id'][0], data['survey_id_count']) for data in survey_data)
+        for survey in self:
+            survey.tot_comp_survey = result.get(survey.id, 0)
+
+    @api.multi
+    def _compute_public_url(self):
+        """ Computes a public URL for the survey """
+        base_url = '/' if self.env.context.get('relative_url') else self.env['ir.config_parameter'].get_param('web.base.url')
+        for survey in self:
+            survey.public_url = urljoin(base_url, "survey/start/%s" % slug(survey))
+
+    @api.multi
+    def _compute_public_url_html(self):
+        """ Computes a public URL for the survey (html-embeddable version)"""
+        self._compute_public_url()
+        for survey in self:
+            survey.public_url_html = '<a href="%s">%s</a>' % (survey.public_url, _("Click here to start survey"))
+
+    @api.multi
+    def _compute_print_url(self):
+        """ Computes a printing URL for the survey """
+        base_url = '/' if self.env.context.get('relative_url') else self.env['ir.config_parameter'].get_param('web.base.url')
+        for survey in self:
+            survey.print_url = urljoin(base_url, "survey/print/%s" % slug(survey))
+
+    @api.multi
+    def _compute_result_url(self):
+        """ Computes an URL for the survey results """
+        base_url = '/' if self.env.context.get('relative_url') else self.env['ir.config_parameter'].get_param('web.base.url')
+        for survey in self:
+            survey.result_url = urljoin(base_url, "survey/results/%s" % slug(survey))
+
+    @api.multi
+    def _read_group_stage_ids(self, domain, read_group_order=None, access_rights_uid=None):
+        """ Read group customization in order to display all the stages in the
+        kanban view, even if they are empty """
+        SurveyStage = self.env['survey.stage']
+        order = SurveyStage._order
+        access_rights_uid = access_rights_uid or self.env.user.id
+
+        if read_group_order == 'stage_id desc':
+            order = '%s desc' % order
+
+        stage_ids = SurveyStage._search([], order=order, access_rights_uid=access_rights_uid)
+        stage = SurveyStage.browse(stage_ids)
+        result = stage.sudo(access_rights_uid).name_get()
+
+        # restore order of the search
+        result.sort(lambda x, y: cmp(stage_ids.index(x[0]), stage_ids.index(y[0])))
+
+        fold = {}
+        for stage in SurveyStage.sudo(access_rights_uid).browse(stage_ids):
```

Comment:
No need of re-browse, you are already browsing stage_ids at line 163(browse it with access_rights_uid and use it here)